great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -178,18 +203,15 @@ def add_year_and_month_splitter(
         self,
         column_name: str,
     ) -> _SQLAsset:
-        """Associates a year month splitter with this DataAsset
-
+        """Associates a year month splitter with this _SQLAsset
         Args:
             column_name: A column name of the date column where year and month will be parsed out.
-
         Returns:
-            This DataAsset so we can use this method fluently.
+            This _SQLAsset so we can use this method fluently.
         """
-        self.column_splitter = SqlYearMonthSplitter(
-            column_name=column_name,
+        raise NotImplementedError(
```

Comment:
If we want to avoid `isinstance` checks in superclasses `_SQLAsset` and `ColumnSplitter` we need to make this abstract and implement in the subclasses due to the diverging logic for `test_connection` on splitters.

Reply:
This has the added benefit of narrowing types.